### PR TITLE
doc: doxygen: improve look and feel

### PR DESCRIPTION
**Pull request by gmarull**
This set of patches improve the look and feel of Doxygen API documentation by using the https://github.com/jothepro/doxygen-awesome-css theme. Doxygen has also been upgraded to 1.9.1 since the theme works better with 1.9.1 or 1.9.2, even though it still has some issues with 1.9.2 on mobile due to changes in navigation bar. Note that, however,  older versions such as 1.8.17 (shipped with Ubuntu LTS) should still work fine to compile the documentation.

As a side note, I think the overall quality of Doxygen documentation should be improved, specially regarding grouping. The advantage of Doxygen docs is that they are fast to build, they are complete and search works well, so it's a valuable piece that isn't promoted enough.

Live test: https://teslabs.github.io/zephyr-doxygen-test/

Before:
image
image

After (light):
image
image

After (dark):
image
image

### CODE REVIEW

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by henrikbrixandersen** [comment]
This is a huge improvement, thank you.

**Review by carlescufi — APPROVED** [review]
The change is literally night and day (no pun intended) when compared to the current theme.
Thank you @gmarull for putting this together!

**Comment by cfriedt** [comment]
MIT is an approved license, fwiu, so the scancode failure seems like a false positive.

**Review by cfriedt — APPROVED** [review]

**Comment by nashif** [comment]
It sure looks like the main zephyr docs, but to be honest, the existing API documentation in the current layout is easier on the eyes and looks smoother, see for example:

image

vs

image


Maybe we need a few more tweaks? What about taking the same colors for variables, i.e. to differentiate them from other bold headers?

**Comment by gmarull** [comment]
@nashif Updated (check live preview, too):

image

**Comment by gmarull** [comment]
Note: I don't think we should resolve these issues here but rather upstream, since these are copied files:
```
-:127: ERROR:TRAILING_WHITESPACE: trailing whitespace
#127: FILE: doc/_doxygen/doxygen-awesome-darkmode-toggle.js:43:
+        // It is possible that the appearance was changed in another tab $

-:152: ERROR:TRAILING_WHITESPACE: trailing whitespace
#152: FILE: doc/_doxygen/doxygen-awesome-darkmode-toggle.js:68:
+        return (!DoxygenAwesomeDarkModeToggle.systemPreference && localStorage.getItem(DoxygenAwesomeDarkModeToggle.prefersDarkModeInLightModeKey)) || $

-:593: WARNING:TYPO_SPELLING: 'collapsable' may be misspelled - perhaps 'collapsible'?
#593: FILE: doc/_doxygen/doxygen-awesome.css:153:
+    /* height of an item in any tree / collapsable table */

-:611: ERROR:TRAILING_WHITESPACE: trailing whitespace
#611: FILE: doc/_doxygen/doxygen-awesome.css:171:
+        $
```

**Comment by nashif** [comment]
@gmarull can we also add some indentation to "See also":

image

**Comment by gmarull** [comment]
> @gmarull can we also add some indentation to "See also":
> 
> image

fixed

**Review by tejlmand — APPROVED** [review]

File: .github/workflows/doc-build.yml [hunk 1]
```diff
@@ -42,7 +43,10 @@ jobs:
 
     - name: install-pkgs
       run: |
-        sudo apt-get install -y ninja-build doxygen graphviz
+        sudo apt-get install -y ninja-build graphviz libclang1-9 libclang-cpp9
```
Comment (tejlmand) [inline .github/workflows/doc-build.yml:46]: curious.
What relation is there to `clang` in this commit ?

That is not clear at all from the commit message, and someone looking at git history in future to figure out why clang was added here will have a hard time to figure it out.

A few words please.

**Review by tejlmand — CHANGES_REQUESTED** [review]
please add some description in the commit message of why clang is now needed for this.

File: .github/workflows/doc-build.yml (hunk 1 above)
Comment (gmarull) [reply .github/workflows/doc-build.yml:46]: Updated commit message, these are Doxygen runtime dependencies.

**Review by MaureenHelm — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]
lgtm

**Pull request merged**